### PR TITLE
Features: Set defaults from manifest on boot

### PR DESCRIPTION
**Pull request by pahearn73**
## What

What does this PR add or remove?

## Why

Why are these changes needed?

## How

How do these changes achieve the goal?

## Test

How has this been tested? How can a reviewer test it?

## Checklist

- [ ] I have self-reviewed this PR
- [ ] I have added tests that prove the feature works or the fix is effective

### CODE REVIEW

File: core/main/src/firebolt/firebolt_ws.rs [hunk 1]
```diff
@@ -221,9 +221,6 @@ impl FireboltWs {
             return;
         }
         if !gateway_secure {
-            state
-                .session_state
-                .add_pending_session(app_id.clone(), None);
```
Comment (pahearn73) [inline core/main/src/firebolt/firebolt_ws.rs:226]: Unrelated but pending sessions are added when perms are checked upon app launch, not needed here.

File: core/main/src/service/context_manager.rs [hunk 2]
```diff
@@ -93,6 +94,14 @@ impl ContextManager {
...
         // Asynchronously get context and update the state
         tokio::spawn(async move {
+            // Set default cloud permissions value
+            ps_c.get_client().get_extn_client().context_update(
+                RippleContextUpdateRequest::UpdateFeatures(vec![FeatureUpdate::new(
+                    FEATURE_CLOUD_PERMISSIONS.into(),
+                    ps_c.get_device_manifest().get_features().cloud_permissions,
+                )]),
+            );
+
```
Comment (pahearn73) [inline core/main/src/service/context_manager.rs:104]: Set default cloud_permissions feature when ContextManager initially starts up, may eventually be overridden by RFC flag.

File: core/main/src/state/cap/permitted_state.rs [hunk 3]
```diff
@@ -150,11 +150,15 @@ impl PermissionHandler {
         app_id: &str,
         allow_cached: bool,
     ) -> RippleResponse {
+        if state.open_rpc_state.is_app_excluded(app_id) {
+            return Ok(());
+        }
+
```
Comment (pahearn73) [inline core/main/src/state/cap/permitted_state.rs:156]: I actually recently added this for just `device_fetch_and_store`, but moved it here so it works for both cloud and device perms.

File: core/sdk/src/api/context.rs [hunk 4]
```diff
@@ -141,14 +141,25 @@ impl RippleContext {
                 false
                 // This is not an update request so need not to honour it
             }
-            RippleContextUpdateRequest::Features(features) => {
+            RippleContextUpdateRequest::UpdateFeatures(features) => {
```
Comment (pahearn73) [inline core/sdk/src/api/context.rs:144]: Improved this a bit so we can pass in one or more features and indicate whether or not they're en/disabled.

**Review by satlead — APPROVED** [review]

**Review by kpears201 — APPROVED** [review]

**Pull request merged**